rdata.c

```diff
@@ -642,6 +647,210 @@ rdata_loc_to_string(buffer_type *ATTR_UNUSED(output),
 	return 0;
 }
 
+static void
+buffer_print_svcparamkey(buffer_type *output, uint16_t svcparamkey)
+{
+	if (svcparamkey < SVCPARAMKEY_COUNT)
+		buffer_printf(output, "%s", svcparamkey_strs[svcparamkey]);
+	else
+		buffer_printf(output, "key%d", (int)svcparamkey);
+}
+
+static int
+rdata_svcparam_port_to_string(buffer_type *output, uint16_t val_len,
+	uint16_t *data)
+{
+	if (val_len != 2)
+		return 0; /* wireformat error, a short is 2 bytes */
+	buffer_printf(output, "=%d", (int)ntohs(data[0]));
+	return 1;
+}
+
+static int
+rdata_svcparam_ipv4hint_to_string(buffer_type *output, uint16_t val_len,
+	uint16_t *data)
+{
+	char ip_str[INET_ADDRSTRLEN];
+	
+	assert(val_len > 0); /* Guaranteed by rdata_svcparam_to_string */
+
+	if ((val_len % IP4ADDRLEN) == 0) {
+		buffer_printf(output, "=%s", inet_ntop(AF_INET, data, ip_str, sizeof(ip_str)));
```

Comment:
The function inet_ntop can return NULL, you have to check the return value from it. Also in line 682.